skimage/segmentation/random_walker_segmentation.py

```diff
@@ -467,6 +467,10 @@ def random_walker(data, labels, beta=130, mode='cg_j', tol=1.e-3, copy=True,
         if data.ndim == 3:  # 2D multispectral, needs singleton in 3rd axis
             data = data[:, :, np.newaxis, :]
 
+    if np.all(labels <= 0):
```

Comment:
This check might a bit expensive at this position. I think we can have it "for free" if we check for `nlabels == 0` after

https://github.com/scikit-image/scikit-image/blob/6423c5f81167e9b673583d6d37d9fff2b8fc84e7/skimage/segmentation/random_walker_segmentation.py#L476-L477

Reply:
it can even be checked within the `_preprocess` function to exit ASAP here
https://github.com/scikit-image/scikit-image/blob/6423c5f81167e9b673583d6d37d9fff2b8fc84e7/skimage/segmentation/random_walker_segmentation.py#L209
if `max(label_values) <= 0`

Reply:
True. Though why not `label_values.size == 1`?

Reply:
`label_values` can be all positives, ie without seed points :)

Reply:
Ah, from `random_walker`'s docstring

> Array of seed markers labeled with different positive integers for different phases. Zero-labeled pixels are unlabeled pixels. Negative labels correspond to inactive pixels that are not taken into account (they are removed from the graph).

Then I'd say check `pos_mask` for labels? https://github.com/scikit-image/scikit-image/blob/6423c5f81167e9b673583d6d37d9fff2b8fc84e7/skimage/segmentation/random_walker_segmentation.py#L223

I'd advocate keeping the "happy path" free of unnecessary checks within reason. Breaking earlier for the unhappy path is nice but not worth the trade-off in my opinion. Though, at this point I'm probably bike-shedding about some minor performance advantage. :sweat_smile:

Reply:
Checking `pos_mask` is almost equivalent to @stefanv proposition (minus `labels <= 0` construction) since we have to check for all the array's value ;). Checking `labels_values` only evaluates few unique label values...
Keep your original proposition if the early exit doesn't worth it :)

Reply:
Makes sense. Thank you for the patience @rfezzani. I'm happy with either suggestion. :+1:

Reply:
I opted for @rfezzani's version. Mine had the problem that `nlabels` is actually `None` if all unlabeled pixels are isolated. In that case `nlabels` does not give any information whether seed points are present or not.